### ISSUE TITLE
FeatureFinderMetaboIdent seems to add a proton to input molecular formulas

### ISSUE THREAD

**Issue by YasinEl** [OpenMS/OpenMS#7069]
It seems that when using FeatureFinderMetaboIdent the masses calculated from the formulas are increased by a proton? 

for example when I give 
```
CompoundName	SumFormula	Mass	Charge	RetentionTime	RetentionTimeRange	IsoDistribution
Sulfadimethoxine	C12H14N4O4S	0	1	198	0	0
```

as tsv input. I get a featureXML with:
```
<UserParam type="string" name="native_id" value="Sulfadimethoxine_m310.073578_z1_rt198.0_i0"/>
<UserParam type="float" name="MZ" value="311.080853643370972"/>
```

However, the `C12H14N4O4S` should rather give `310.07303` with a `1` charge. Is this behavior intended? If yes how could I change it if I want to extract adducts other than M+H (like e.g. M+Na)? 

Thank you!

**Comment by timosachsenberg** [comment]
@hendrikweisser any comment here?

**Comment by jpfeuffer** [comment]
Without knowing the code in detail, after a short initial analysis, I would say that FFMIdent does not support different adducts, yet.
If I am not mistaken, one could probably model this by adding versions of the compound to the input list with an elemental formula of `compound + adduct` and a given mass of `compound + adduct - charge * PROTON_MASS`

But an internal handling would of course be nicer and more exact

**Comment by YasinEl** [comment]
Yes, that is indeed how I am currently solving the issue on my side, but wanted to know if there is another way since it does not feel like a very clean solution. 
Thank you for confirming!

**Comment by hendrikweisser** [comment]
I just got back from holiday. I can confirm that protons are automatically added for the charges, and that the tool would need to be extended to support other adducts. (We have support for adducts e.g. in NucleicAcidSearchEngine already...)

**Comment by YasinEl** [comment]
I am now subtracting the proton from each input formula and then add the actual adduct I want to the input formula. Maybe as a small feature request: Allowing to simply subtract/add an electron for every charge in the charge column would allow users to define any adduct they like. 
Anyway, thank you!
